modules/base/rendering/grids/renderableradialgrid.cpp

```diff
@@ -0,0 +1,461 @@
+}
+
+void RenderableRadialGrid::render(const RenderData& data, RendererTasks&){
+    _gridProgram->activate();
+
+    _gridProgram->setUniform("opacity", _opacity);
+
+    const glm::dmat4 modelTransform =
+        glm::translate(glm::dmat4(1.0), data.modelTransform.translation) * // Translation
+        glm::dmat4(data.modelTransform.rotation) *  // Spice rotation
+        glm::scale(glm::dmat4(1.0), glm::dvec3(data.modelTransform.scale));
+
+    const glm::dmat4 modelViewTransform = data.camera.combinedViewMatrix() *
+                                          modelTransform;
+
+    _gridProgram->setUniform("modelViewTransform", modelViewTransform);
+    _gridProgram->setUniform(
+        "MVPTransform",
+        glm::dmat4(data.camera.projectionMatrix()) * modelViewTransform
+    );
+    
+    _gridProgram->setUniform("gridColor", _gridColor);
+
+    float adjustedLineWidth = 1.f;
+
+#ifndef __APPLE__
+    adjustedLineWidth = _lineWidth;
+#endif
+
+    // Saves current state:
+    GLboolean isBlendEnabled = glIsEnabledi(GL_BLEND, 0);
+    GLfloat currentLineWidth;
+    glGetFloatv(GL_LINE_WIDTH, &currentLineWidth);
+    GLboolean isLineSmoothEnabled = glIsEnabled(GL_LINE_SMOOTH);
+
+    GLenum blendEquationRGB;
+    GLenum blendEquationAlpha;
+    GLenum blendDestAlpha;
+    GLenum blendDestRGB;
+    GLenum blendSrcAlpha;
+    GLenum blendSrcRGB;
+    glGetIntegerv(GL_BLEND_EQUATION_RGB, &blendEquationRGB);
+    glGetIntegerv(GL_BLEND_EQUATION_ALPHA, &blendEquationAlpha);
+    glGetIntegerv(GL_BLEND_DST_ALPHA, &blendDestAlpha);
+    glGetIntegerv(GL_BLEND_DST_RGB, &blendDestRGB);
+    glGetIntegerv(GL_BLEND_SRC_ALPHA, &blendSrcAlpha);
+    glGetIntegerv(GL_BLEND_SRC_RGB, &blendSrcRGB);
+
+    // Changes GL state:
+    glLineWidth(adjustedLineWidth);
+    glEnablei(GL_BLEND, 0);
+    glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+    glEnable(GL_LINE_SMOOTH);
+    
+    for (GeometryData &c : _circles) {
+        c.render();
+    }
+
+    _lines.render();
+
+    _gridProgram->deactivate();
+
+    // Restores GL State
+    glLineWidth(currentLineWidth);
+    glBlendEquationSeparate(blendEquationRGB, blendEquationAlpha);
+    glBlendFuncSeparate(blendSrcRGB, blendDestRGB, blendSrcAlpha, blendDestAlpha);
+
+    if (!isBlendEnabled) {
+        glDisablei(GL_BLEND, 0);
+    }
+
+    if (!isLineSmoothEnabled) {
+        glDisable(GL_LINE_SMOOTH);
+    }
+}
+
+void RenderableRadialGrid::update(const UpdateData&) {
+    if (_gridIsDirty) {
+
+        // Circles
+        const int nRadialSegments = _gridSegments.value()[0];
+        const float fnCircles = static_cast<float>(nRadialSegments);
+        const float deltaRadius = (_maxRadius - _minRadius) / fnCircles;
+
+        const bool hasInnerRadius = _minRadius > 0;
+        const int nCircles = hasInnerRadius ? nRadialSegments : nRadialSegments + 1;
+
+        _circles.clear();
+        _circles.reserve(nCircles);
+
+        auto addRing = [&](int nSegments, float radius) {
```

Comment:
Instead of `&`, you can capture `&_circles` directly and make it clear which variables are captured

Reply:
That doesn't seem to work unfortunately, since `_circles` is a private member (?). Get the error that it's not a variable. 

Would it be clearer to capture `this` instead? Or maybe do something like: `&circles = this->_circles`

Reply:
@alexanderbock ? :)

Reply:
Decided to go with `this`, as the other option looked a bit messy to me :) (did not wait for respond since I wanted this branch merged to do the cleanup in my other PR)